x/ibc/02-client/handler.go

```diff
@@ -104,3 +106,48 @@ func HandlerClientMisbehaviour(k keeper.Keeper) evidencetypes.Handler {
 		return nil
 	}
 }
+
+// HandleClientUpdateProposal is a handler for executing a passed client update proposal
+func HandleClientUpdateProposal(ctx sdk.Context, k keeper.Keeper, p *ibctypes.ClientUpdateProposal) error {
+	clientState, found := k.GetClientState(ctx, p.ClientId)
+	if !found {
+		return types.ErrClientNotFound
+	}
+
+	clientType := clientState.ClientType()
+	switch clientType {
+	case exported.Tendermint:
+		tmClientState := clientState.(*ibctmtypes.ClientState)
+
+		updateClientFlag := false
+		if tmClientState.AllowGovernanceOverrideAfterExpiry && tmClientState.Expired() {
+			updateClientFlag = true
+		}
+
+		if tmClientState.AllowGovernanceOverrideAfterMisbehaviour && tmClientState.IsFrozen() {
```

Comment:
This is incorrect behaviour from my understanding.

Suppose `AllowGovernanceOverrideAfterExpiry = false` and `AllowGovernanceOverrideAfterMisbehaviour = true`

And the client is frozen AND expired, then the correct behaviour is to reject the proposal since client is still expired even if it gets unfrozen.

Similarly, when `AllowGovernanceOverrideAfterExpire = true` and `AllowGovernanceOverrideAfterMisbehaviour = false`

And the client is frozen AND expired, then the update proposal should fail since the client is still frozen.

cc: @cwgoes

Reply:
This code will allow the proposal to pass in the above cases

Reply:
When we change this above, we should end up with two separate paths (`UpdateClient` will be called here, but not in the above path)